### PR TITLE
Improve error messages for failed plugin loads

### PR DESCRIPTION
**Pull request by dann-merlin**
This improves the error messages you get when elvish fails to load a go module.
Currently elvish almost always just fails with a NoSuchModule error, even when the error was actually a library mismatch or a missing Ns symbol in the .so file.

### CODE REVIEW

**Comment by dann-merlin** [comment]
Forgot to run the linters/tests. Updated it now.

File: pkg/eval/builtin_special.go [hunk 1]
```diff
@@ -41,8 +41,17 @@ var IsBuiltinSpecial = map[string]bool{}
 // NoSuchModule encodes an error where a module spec cannot be resolved.
 type NoSuchModule struct{ spec string }
 
+// ModuleLoadError encapsulates an error message a failed module load operation.
+type ModuleLoadError struct {
+	spec     string
+	errormsg string
```
Comment (xiaq) [inline pkg/eval/builtin_special.go:47]: Wrapper errors in Go usually store the original `error` and implement the `Unwrap` method as documented in https://pkg.go.dev/errors.

File: pkg/eval/builtin_special.go [hunk 2]
```diff
@@ -41,8 +41,17 @@ var IsBuiltinSpecial = map[string]bool{}
...
+}
+
 // Error implements the error interface.
 func (err NoSuchModule) Error() string { return "no such module: " + err.spec }
+func (err ModuleLoadError) Error() string {
+	return "when trying to load module: \"" + err.spec + "\" this error occurred: " + err.errormsg
```
Comment (xiaq) [inline pkg/eval/builtin_special.go:53]: Elvish's codebase follows this convention: https://go.dev/wiki/CodeReviewComments#error-strings. I'd write this error message as simply:

```go
"load module " + err.spec + ": " + err.errmsg
```

File: pkg/eval/builtin_special.go [hunk 3]
```diff
@@ -41,8 +41,17 @@ var IsBuiltinSpecial = map[string]bool{}
 // NoSuchModule encodes an error where a module spec cannot be resolved.
 type NoSuchModule struct{ spec string }
 
+// ModuleLoadError encapsulates an error message a failed module load operation.
+type ModuleLoadError struct {
```
Comment (xiaq) [inline pkg/eval/builtin_special.go:45]: This should probably be `PluginLoadError` because it's not used for Elvish modules.

**Comment by xiaq** [comment]
Thanks, there are some style issues that I'll fix up after merging.

**Pull request merged**